olga/analytics/views.py

```diff
@@ -48,32 +121,66 @@ def update_students_without_country_value(active_students_amount, students_per_c
             students_per_country (dict): Country-count accordance as pair of key-value.
                                          Amount of students without country has calculated and inserted to
                                          corresponding key ('null').
-        """
 
+        """
         students_per_country['null'] = active_students_amount - sum(students_per_country.values())
 
         return students_per_country
 
-    def create_instance_data(self, received_data, secret_token):  # pylint: disable=too-many-locals
+    def extend_statistics_to_enthusiast_level(self, received_data, installation_statistics, edx_installation_object):
+        # pylint: disable=invalid-name
+        """
+        Extend installation statistics level from `Paranoid` to `Enthusiast`.
+
+        It contains additional information about installation:
+            - latitude, longitude;
+            - platform name, platform url;
+            - students per country.
+        """
+        active_students_amount_day = int(received_data.get('active_students_amount_day'))
+
+        # Decoded to process and encoded to save in database list of dictionaries,
+        # that contains amount of students per country
+        students_per_country_decoded = json.loads(received_data.get('students_per_country'))
+        students_per_country_encoded = json.dumps(self.update_students_without_country_value(
+            active_students_amount_day, students_per_country_decoded
+        ))
+
+        enthusiast_edx_installation = {
+            'latitude': float(received_data.get('latitude')),
+            'longitude': float(received_data.get('longitude')),
+            'platform_name': received_data.get('platform_name'),
+            'platform_url': received_data.get('platform_url'),
+        }
+
+        enthusiast_installation_statistics = {  # pylint: disable=invalid-name
+            'students_per_country': students_per_country_encoded
+        }
+
+        installation_statistics.update(enthusiast_installation_statistics)
```

Comment:
`stats` as `data` or `manager` are obscure names. Statistics belongs to something and it need to be displayed, IMHO.